var/spack/repos/builtin/packages/slepc/package.py

```diff
@@ -160,31 +162,43 @@ def archive_files(self):
         return [join_path(self.stage.source_path, 'configure.log'),
                 join_path(self.stage.source_path, 'make.log')]
 
-    def run_hello_test(self):
-        """Run stand alone test: hello"""
-        test_dir = self.test_suite.current_test_data_dir
-
-        if not os.path.exists(test_dir):
-            print('Skipping slepc test')
-            return
+    @run_after('install')
+    def setup_build_tests(self):
+        """Copy the build test files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources([join_path('src', 'eps', 'tests')])
 
-        exe = 'hello'
+    def run_test1_example(self, test_dir):
+        exe = 'test1'
         cc_exe = os.environ['CC']
 
+        self.run_test(exe='make',
+                      options=[join_path(test_dir, 'makefile'), exe],
+                      purpose='test:compile makefile',
+                      work_dir=test_dir)
+
         self.run_test(exe=cc_exe,
                       options=['-I{0}'.format(self.prefix.include),
-                               '-L', self.prefix.lib, '-l', 'slepc',
-                               '-L', self.spec['petsc'].prefix.lib, '-l', 'petsc',
-                               '-L', self.spec['mpi'].prefix.lib, '-l', 'mpi',
-                               '-o', exe, join_path(test_dir, 'hello.c')],
+                               '-L{0}'.format(self.prefix.lib),
+                               '-L{0}'.format(self.spec['petsc'].prefix.lib),
+                               '-L{0}'.format(self.spec['mpi'].prefix.lib),
+                               join_path(test_dir, 'test1.c'), '-o', exe,
+                               '-lslepc', '-lpetsc', '-lmpi', '-lm'],
                       purpose='test: compile {0} example'.format(exe),
                       work_dir=test_dir)
 
-        self.run_test(exe=exe,
-                      options=[],
-                      expected=['Hello world'],
+        self.run_test(exe,
```

Comment:
Did you want to conditionally perform this check -- or warn that it is skipped -- based on the results from `make` as you've started doing in other PRs?